probe_src/python/probe_py/manual/remote_access.py

```diff
@@ -0,0 +1,622 @@
+import traceback
+import dataclasses
+import base64
+from typing import Union, Any
+
+from probe_py.manual.persistent_provenance import (
+    InodeVersion,
+    Inode,
+    get_prov_upstream,
+    Process,
+    InodeMetadataVersion,
+)
+import struct
+import xdg_base_dirs
+import random
+import pickle
+import datetime
+import json
+import re
+import os
+import socket
+import subprocess
+import pathlib
+import typer
+import yaml
+
+PROBE_HOME = xdg_base_dirs.xdg_data_home() / "PROBE"
+PROCESS_ID_THAT_WROTE_INODE_VERSION = PROBE_HOME / "process_id_that_wrote_inode_version"
+PROCESSES_BY_ID = PROBE_HOME / "processes_by_id"
+
+
+
+def copy(cmd: list[str])->None:
+    # 1. get the src_inode_version and src_inode_metadata
+    # 2. upload the files
+    # 3. get process closure and file_writes
+    # 4. get remote home for dest
+    # 5. create the two directories on dest
+    # 6. upload src_inode version to the destination
+    # 7. upload process that wrote the file, file to dest
+    #  do this for the dest
+    # 8. generate random_pid to refer to the process
+    # 9. create an scp process json
+    # 10. added reference to scp process id to the dest inode version
+    # 11. copy the scp process to the file on dest
+    try:
+        # Extract port from SCP command
+        port = extract_port_from_scp_command(cmd)
+        if port is None:
+            port = 22
```

Comment:
This port is for the other scp command we use to transfer the process-by-id and process_id_that_wrote_inode_version files. I believe all options of the main scp command will apply to the file transfer except for -r. Instead of including this option, I should collect the scp_options and use them to run the subsequent scp commands.

Reply:
@charmoniumQ